components/kyma-environment-broker/internal/process/provisioning/vmuser_overrides.go

```diff
@@ -0,0 +1,125 @@
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/broker"
+	kebError "github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/error"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/process"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/vmauthoverrides"
+	"github.com/kyma-project/control-plane/components/provisioner/pkg/gqlschema"
+	"github.com/sirupsen/logrus"
+
+	"helm.sh/helm/v3/pkg/release"
+)
+
+const (
+	VMAuthComponentName = "runtime-monitoring-agent"
+	ProfileName         = "values"
+)
+
+type HelmClient interface {
+	ListReleases(name string) ([]*release.Release, error)
+	InstallRelease(chartDir, releaseName string, namespace string, overrides map[string]interface{}, profile string) (*release.Release, error)
+}
+
+type VMAuthOverridesStep struct {
+	operationManager *process.ProvisionOperationManager
+	client           HelmClient
+	vmAuthConfig     vmauthoverrides.Config
+}
+
+func NewVMAuthOverridesStep(os storage.Operations, client HelmClient, vmAuthConfig vmauthoverrides.Config) *VMAuthOverridesStep {
+	return &VMAuthOverridesStep{
+		operationManager: process.NewProvisionOperationManager(os),
+		client:           client,
+		vmAuthConfig:     vmAuthConfig,
+	}
+}
+
+func (s *VMAuthOverridesStep) Name() string {
+	return "VM_Auth_Overrides"
+}
+
+func (s *VMAuthOverridesStep) Run(operation internal.ProvisioningOperation, log logrus.FieldLogger) (internal.ProvisioningOperation, time.Duration, error) {
+	releaseName := operation.InstanceDetails.ShootName
+	isPresent := false
+	releases, err := s.client.ListReleases(s.vmAuthConfig.Namespace)
+	for _, rel := range releases {
+		if rel.Name == releaseName {
+			isPresent = true
+		}
+	}
+	if err != nil {
+		return s.handleError(operation, err, "failed to check release existence", log)
+	}
+
+	vmPassword := vmauthoverrides.GeneratePassword(isPresent, operation)
+	if !isPresent {
+		log.Info("Release not found. Ready to install.")
+		overrideMap := s.generateOverrideMap(operation, vmPassword)
+
+		_, err = s.client.InstallRelease(s.vmAuthConfig.ChartUrl, releaseName, s.vmAuthConfig.Namespace, overrideMap, ProfileName)
+		if err != nil {
+			return s.handleError(operation, err, "failed to install chart", log)
+		}
+	}
+
+	log.Info("Override username and password")
+	VMAuthOverrides := []*gqlschema.ConfigEntryInput{
+		{
+			Key:   "vmuser.username",
+			Value: operation.InstanceID,
+		},
+		{
+			Key:   "vmuser.password",
+			Value: vmPassword,
+		},
+	}
+	operation.InputCreator.AppendOverrides(VMAuthComponentName, VMAuthOverrides)
+
+	retry := time.Duration(0)
+	operation, retry = s.operationManager.UpdateOperation(operation, func(operation *internal.ProvisioningOperation) {
```

Comment:
We should only save the monitoring data when a new password is generated, updating the record at every try could lead to infinite operation processing